### ISSUE TITLE
Add release cycle support to `schedule-builder`

### ISSUE THREAD

**Issue by saschagrunert** [kubernetes/release#2207]
We do not have a template for creating new release cycles ([like v1.23](https://github.com/kubernetes/sig-release/blob/master/releases/release-1.23/README.md)), which means that we copy the process from the previous document to the next one.

The idea is to add a new functionality to [`schedule-builder`](https://github.com/kubernetes/release/tree/master/cmd/schedule-builder), which allows to create new release cycles.

This would also allow us to put the release schedule on k/website later on if we want to go this route.

**Comment by mritunjaysharma394** [comment]
Understanding it a little bit, I have realised that this mainly require changes in [model.go](https://github.com/kubernetes/release/blob/eeb22d265892bbe2aa2ad3abc1e581f0c1630e79/cmd/schedule-builder/cmd/model.go#L20) like this below where in addition to Patch Schedule we can have a Release schedule struct? https://github.com/kubernetes/release/blob/eeb22d265892bbe2aa2ad3abc1e581f0c1630e79/cmd/schedule-builder/cmd/model.go#L20

**Comment by saschagrunert** [comment]
@mritunjaysharma394 yes I think adding a new data model could be a good starting point. This model has to be transferred into a template which should look similar to: https://github.com/kubernetes/sig-release/blob/master/releases/release-1.23/README.md

For example, a `timeline` could be a slice of `events` happening during the release, whereas an `event` represents the `What | Who | When | WEEK | CI SIGNAL` from the table.

**Comment by mritunjaysharma394** [comment]
Thanks a lot for the suggestion @saschagrunert, I will try to create a WIP PR today to understand and learn more deeply, thank you for the help! 

/assign
